iap_alternative_provider/models/iap_account.py

```diff
@@ -12,20 +12,18 @@ class IapAccount(models.Model):
 
     def _get_service_from_provider(self):
         """In case that the provider only propose one service you can
-        return the service_name in you module to simplify the user interface"""
+        return the service_id in your module to simplify the user interface"""
         return None
 
     def _set_service_from_provider(self):
         for record in self:
             service = record._get_service_from_provider()
-            if service and record.service_name != service:
-                record.service_name = service
-
-    @api.model_create_multi
-    def create(self, vals_list):
```

Comment:
should keep this method in the case a record of iap.account created from `csv` or `xml` file